conformance/utils/http/http.go

```diff
@@ -344,14 +344,27 @@ func setRedirectRequestDefaults(req *roundtripper.Request, cRes *roundtripper.Ca
 		expected.RedirectRequest.Host = cRes.RedirectRequest.Host
 	}
 
-	if expected.RedirectRequest.Port == "" {
-		expected.RedirectRequest.Port = req.URL.Port()
-	}
-
 	if expected.RedirectRequest.Scheme == "" {
 		expected.RedirectRequest.Scheme = req.URL.Scheme
 	}
 
+	if expected.RedirectRequest.Port == "" {
+		func() {
```

Comment:
Also, why wrap this in an anonymous function? I don't see the advantage?

Reply:
Perfectly valid question. Maybe it was just a personal preference - I felt that it made the code more readable and matched the way in which we've described this condition in our API reference:
<img width="538" alt="image" src="https://user-images.githubusercontent.com/26463869/229346710-bf1326fc-0cca-4f0f-ba1d-1168128bdab0.png">


```
If http with port 80 -> do nothing

If https with 443 -> do nothing

If none of the previous conditions -> set default value.
```

But yeah, I've changed it to a different format while still trying to avoid complex/long if/else. Hope this is more generally appealing.